### PR TITLE
Add new replication parameter 'mode' to storage class

### PR DESCRIPTION
**Pull request by santhoshatdell**
# Description
Added new replication parameter 'mode' to storage class. The replication mode values can be sync/async/metro.
Default is 'Async' to support backwards compatibility.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1443 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works (More tests will be added when create metro volume is implemented)
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Performed volume creations with just the driver and made sure that the new code works as expected and backward compatibility is not broken.
PV without any replication parameters created successfully.

Driver log:
#### when invalid mode is specified in the Storage Class
Warning  ProvisioningFailed    14s (x7 over 77s)  csi-powerstore.dellemc.com_powerstore-controller-567bc8ff4c-8qxmf_3b81a379-e9ec-4831-b6c0-370f1ae08627  failed to provision volume with StorageClass "powerstore-replication-invalid": rpc error: code = InvalidArgument desc = replication enabled but invalid replication mode specified in storage class
  
#### when Metro mode is specified in the Storage Class
{"level":"info","msg":"Preparing volume replication","time":"2024-09-03T16:33:42.866888418Z"}
{"level":"info","msg":"Metro replication mode requested","time":"2024-09-03T16:33:42.866908807Z"}

#### when no mode or Async is specified in the Storage Class
{"level":"info","msg":"Preparing volume replication","time":"2024-09-03T16:43:02.25396221Z"}
{"level":"info","msg":"ASYNC replication mode requested","time":"2024-09-03T16:43:02.253974847Z"}

### CODE REVIEW

**Review by lukeatdell — APPROVED** [review]

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -249,84 +249,110 @@ func (s *Service) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest
...
+		repMode = strings.ToUpper(repMode)
 
-		namespace := ""
-		if ignoreNS, ok := params[s.WithRP(KeyReplicationIgnoreNamespaces)]; ok && ignoreNS == "false" {
-			pvcNS, ok := params[KeyCSIPVCNamespace]
-			if ok {
-				namespace = pvcNS + "-"
+		switch repMode {
+		case "SYNC", "ASYNC":
+			// handle Sync and Async modes
```
Comment (donatwork) [inline pkg/controller/controller.go:270]: The case statement already indicate that the block handles Sync and Async.

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -249,84 +249,110 @@ func (s *Service) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest
...
+			log.Infof("%s replication mode requested", repMode)
+			vgPrefix, ok := params[s.WithRP(KeyReplicationVGPrefix)]
```
Comment (donatwork) [inline pkg/controller/controller.go:272]: Are we always using volume groups?

**Review by donatwork — APPROVED** [review]

File: pkg/controller/controller.go (hunk 2 above)
Comment (santhoshatdell) [reply pkg/controller/controller.go:272]: Yes. I made a comment in the Metro section to optimize the volume group calls as we add the metro calls in later PRs.

File: pkg/controller/controller.go (hunk 1 above)
Comment (santhoshatdell) [reply pkg/controller/controller.go:270]: We could update the comments later to indicate how different the modes are, w.r.t volume group and rules.

**Pull request merged**